aiven_db_migrate/migrate/pgmigrate.py

```diff
@@ -963,7 +1012,37 @@ def _check_pg_ext(self):
                     dbname
                 )
 
-    def _check_pg_lobs(self):
+    def _check_aiven_pg_security_agent(self) -> None:
+        if not self.target.has_aiven_gatekeeper:
+            return
+
+        self.log.info("Aiven PostgreSQL security agent is enabled in target.")
+
+        # Get the list of roles that are in the source but not in the target.
+        missing_roles = set(self.source.pg_roles.keys()) - set(self.target.pg_roles.keys())
+        # Only the superuser roles.
+        su_missing_roles = {r for r in missing_roles if self.source.pg_roles[r].rolsuper}
+        # The only allowed superuser roles in the target.
+        reserved_roles = set(self.target.get_security_agent_reserved_roles())
+
+        unauthorized_roles = su_missing_roles - reserved_roles
+
+        if unauthorized_roles:
+            self.log.error(
+                "Superuser roles %r from source database are not allowed in target database. "
+                "Allowed roles: %r.",
```

Comment:
nit: I think we should not expose more data in the logs than needed. Throwing an error for unauthorized OK, but I would not log allowed in this case.

Reply:
If it's a security issue, it's not a nitpick.
Who has access to these logs? Probably only people who run this command? Or could also be other people?

Reply:
Logs can be collected and aggregated somewhere. Actually when you ask that, I'd just remove mentioning roles from the logs in general, just leave generic message. Error with info will be propagated to the client anyway.